### PR TITLE
Tagged LongValueOccurence message deserializer

### PR DESCRIPTION
**Pull request by tmasternak**
Connects to https://github.com/Particular/LaunchWave.DevOpsAdvancedMonitoring/issues/167

### CODE REVIEW

File: src/ServiceControl.Monitoring/Messaging/TaggedLongValueWriterOccurrenceSerializerDefinition.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        {
+            var message = RawMessage.Pool<TaggedLongValueOccurrence>.Default.Lease();
+            message.TagValue = keyValue;
+
+            return message;
+        }
+
+        static Dictionary<int, string> DeserializeTags(int tagCount, BinaryReader reader)
+        {
+            var tagDecoder = new UTF8Encoding(false);
```
Comment (Scooletz) [inline src/ServiceControl.Monitoring/Messaging/TaggedLongValueWriterOccurrenceSerializerDefinition.cs]: This should be `static readonly` field

**Comment by tmasternak** [comment]
@Scooletz I think that he answers are `yes` and `yes` :). Let me do fixes and add tests. After that I'll ping maintainers.

File: src/ServiceControl.Monitoring/Messaging/TaggedLongValueWriterOccurrenceSerializerDefinition.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                for (var i = 0; i < count; i++)
+                {
+                    var ticks = reader.ReadInt32();
+                    var timestamp = baseTicks + ticks;
+                    var tagKey = reader.ReadInt32();
+                    var value = reader.ReadInt64();
+
+                    var message = tagKeyToMessage[tagKey];
+
+                    if (message.Length == RawMessage.MaxEntries)
```
Comment (mikeminutillo) [inline src/ServiceControl.Monitoring/Messaging/TaggedLongValueWriterOccurrenceSerializerDefinition.cs]: Would it be worth dropping an `IsFull` property on `RawMessage`?

Comment (tmasternak) [reply src/ServiceControl.Monitoring/Messaging/TaggedLongValueWriterOccurrenceSerializerDefinition.cs]: 👍

**Pull request merged**